bin/datatool/src/args.rs

```diff
@@ -0,0 +1,187 @@
+//! Command line arguments for Strata's `datatool` binary.
+
+use std::path::PathBuf;
+
+use argh::FromArgs;
+use bitcoin::Network;
+use rand::rngs::OsRng;
+
+/// Args.
+#[derive(FromArgs)]
+pub(crate) struct Args {
+    #[argh(option, description = "network name [signet, regtest]", short = 'b')]
+    pub(crate) bitcoin_network: Option<String>,
+
+    #[argh(
+        option,
+        description = "data directory (unused) (default cwd)",
+        short = 'd'
+    )]
+    pub(crate) datadir: Option<PathBuf>,
+
+    #[argh(subcommand)]
+    pub(crate) subc: SubcommandGen,
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+pub(crate) enum SubcommandGen {
+    Seed(SubcSeed),
+    SeqPubkey(SubcSeqPubkey),
+    SeqPrivkey(SubcSeqPrivkey),
+    OpXpub(SubcOpXpub),
+    Params(SubcParams),
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(
+    subcommand,
+    name = "genseed",
+    description = "generates a xpriv and writes it to a file"
+)]
+pub(crate) struct SubcSeed {
+    #[argh(positional, description = "output path")]
+    pub(crate) path: PathBuf,
+
+    #[argh(switch, description = "force overwrite", short = 'f')]
+    pub(crate) force: bool,
+}
```

Comment:
While we're in here we should rename this from "genseed" since it's not actually generating a *seed*, it's directly generating the master key.  So either we should make it do that or rename it.

Reply:
Good point done in 0d73020d7841b9f30112b2a58684f1ab89f3ddf6